app/webpack/components/user_profile/profile.vue

```diff
@@ -0,0 +1,283 @@
+              aria-label="Recipient's display name"
+              aria-describedby="button-updateDisplay"
+            />
+            <div class="input-group-append">
+              <button
+                id="button-updateDisplay"
+                class="btn btn-secondary"
+                type="button"
+                @click="onUpdateName"
+              >
+                {{ updateI18n }}
+              </button>
+            </div>
+            <span class="w-100"></span>
+            <small id="nameHelpBlock" class="form-text text-muted"></small>
+          </div>
+        </div>
+      </div>
+      <div class="form-group row">
+        <label for="inputEmail" class="col-sm-3 col-form-label">
+          {{ emailI18n }}
+        </label>
+        <div class="col-sm-9">
+          <div class="input-group mb-3">
+            <input
+              id="inputEmail"
+              v-model="userEmail"
+              type="email"
+              class="form-control"
+              placeholder="Email Address"
+              aria-label="Recipient's email address"
+              aria-describedby="button-updateEmail"
+            />
+            <div class="input-group-append">
+              <button
+                id="button-updateEmail"
+                class="btn btn-secondary"
+                type="button"
+                @click="onUpdateEmail"
+              >
+                {{ updateI18n }}
+              </button>
+            </div>
+            <span class="w-100"></span>
+            <small id="emailHelpBlock" class="form-text text-muted">
+              {{ emailHelpI18n }}
+            </small>
+          </div>
+        </div>
+      </div>
+      <div class="card bg-light">
+        <div class="card-body">
+          <h5 class="card-title">{{ profileI18n.title }}</h5>
+          <h6 class="card-subtitle mb-2 text-muted">
+            {{ profileI18n.subtitle }}
+          </h6>
+          <p class="card-text">
+            {{ profileI18n.text }}
+          </p>
+          <a :href="deletionMailtoLink" class="card-link">
+            {{ profileI18n.link }}
+          </a>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import firebase, { initializeApp } from 'firebase/app';
+import Axios from 'axios';
+import Turbolinks from 'turbolinks';
+import SignIn from '../authentication/sign_in';
+
+export default {
+  components: {
+    SignIn
+  },
+  props: {
+    apiKey: {
+      type: String,
+      required: true
+    },
+    projectId: {
+      type: String,
+      required: true
+    }
+  },
+  data() {
+    return {
+      loading: true,
+      csrf_token: null,
+      currentUser: null,
+      showSignIn: false,
+      displayName: null,
+      userEmail: null
+    };
+  },
+  computed: {
+    authenticateMessage() {
+      return 'Please sign in again to modify your profile';
+    },
+    pageHeader() {
+      return `Hi there, ${this.currentUser.displayName}`;
+    },
+    displayNameI18n() {
```

Comment:
Doesn't look like you need all these special methods for static strings? Could you put the strings directly in the template? If/when we do add translations support in the UI components then it'll likely have it's own mechanisms for pulling in the right string.